libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -673,13 +673,27 @@ bool AP_AHRS_NavEKF::get_relative_position_NED(Vector3f &vec) const
         return false;
 
 #if AP_AHRS_WITH_EKF1
-    case EKF_TYPE1:
-        return EKF1.getPosNED(vec);
+    case EKF_TYPE1: {
+        Vector2f posNE;
+        float posD;
+        bool position_is_valid = (EKF1.getPosNE(posNE) && EKF1.getPosD(posD));
+        vec.x = posNE.x;
+        vec.y = posNE.y;
+        vec.z = posD;
+        return position_is_valid;
+    }
 #endif
 
     case EKF_TYPE2:
-    default:
-        return EKF2.getPosNED(-1,vec);
+    default: {
```

Comment:
The only reason i can think of is defensive programming whereby an invalid value is sent